rossum_api/elis_api_client_sync.py

```diff
@@ -340,231 +330,272 @@ def poll_annotation(
 
         Sideloading is done only once after the predicate becomes true to avoid spamming the server.
         """
-        return self._run_coroutine(
-            self.elis_api_client.poll_annotation(annotation_id, predicate, sleep_s, sideloads)
+        resource = Resource.Annotation
+
+        annotation_response = self.internal_client.fetch_resource(resource, annotation_id)
+        # Deserialize early, we want the predicate to work with Annotation instances for convenience.
+        annotation = self._deserializer(resource, annotation_response)
+
+        while not predicate(annotation):
+            time.sleep(sleep_s)
+            annotation_response = self.internal_client.fetch_resource(resource, annotation_id)
+            annotation = self._deserializer(resource, annotation_response)
+
+        if sideloads:
+            self.internal_client.sideload(annotation_response, sideloads)
+        return self._deserializer(resource, annotation_response)
```

Comment:
I remember we were not happy about this part during the hackaton. I think we could simplify this at the expense of 1 extra API call at the end (fetching the annotation once more instead of just the sideloads).

```python
        annotation = self.retrieve_annotation(annotation_id)

        while not predicate(annotation):
            time.sleep(sleep_s)
            annotation = self.retrieve_annotation(annotation_id)

        return self.retrieve_annotation(annotation_id, sideloads)